x-pack/plugins/enterprise_search/public/applications/app_search/components/result/result.tsx

```diff
@@ -56,48 +55,26 @@ export const Result: React.FC<Props> = ({
     [result]
   );
   const numResults = resultFields.length;
-  const documentLink = generateEncodedPath(ENGINE_DOCUMENT_DETAIL_PATH, {
-    engineName: resultMeta.engine,
-    documentId: resultMeta.id,
-  });
 
   const typeForField = (fieldName: string) => {
     if (schemaForTypeHighlights) return schemaForTypeHighlights[fieldName];
   };
 
-  const ResultActions = () => {
-    if (!shouldLinkToDetailPage && !actions.length) return null;
-    return (
-      <EuiFlexItem grow={false}>
-        <EuiFlexGroup gutterSize="s">
-          {shouldLinkToDetailPage && (
-            <ReactRouterHelper to={documentLink}>
-              <EuiFlexItem grow={false}>
-                <EuiButtonIcon
-                  iconType="eye"
-                  data-test-subj="DocumentDetailLink"
-                  aria-label={i18n.translate(
-                    'xpack.enterpriseSearch.appSearch.result.documentDetailLink',
-                    { defaultMessage: 'Visit document details' }
-                  )}
-                />
-              </EuiFlexItem>
-            </ReactRouterHelper>
-          )}
-          {actions.map(({ onClick, title, iconType, iconColor }) => (
-            <EuiFlexItem key={title}>
-              <EuiButtonIcon
-                iconType={iconType}
-                onClick={onClick}
-                color={iconColor ? iconColor : 'primary'}
-                aria-label={title}
-              />
-            </EuiFlexItem>
-          ))}
-        </EuiFlexGroup>
-      </EuiFlexItem>
-    );
-  };
+  const documentLink = shouldLinkToDetailPage
+    ? generateEncodedPath(ENGINE_DOCUMENT_DETAIL_PATH, {
+        engineName: resultMeta.engine,
+        documentId: resultMeta.id,
+      })
+    : undefined;
+  if (shouldLinkToDetailPage && documentLink) {
+    actions.unshift({
```

Comment:
@JasonStoltz I was 50/50 on using `unshift` vs spread operator (e.g. `[{ linkAction }, ...action]`) here. I went with unshift for brevity and because mutating the array doesn't super matter in this component but happy to change if you feel strongly otherwise

Reply:
Also just to speak a bit more on this functional change - I do strongly believe that changing the custom view button link to a standard action/button instead is 1. simpler code-wise and 2. actually a better screen reader/title experience. An `<a>` link being present on the document ID is a strong enough indicator, the button doesn't need it as well.

<img width="152" alt="Screen Shot 2021-04-09 at 10 15 17 AM" src="https://user-images.githubusercontent.com/549407/114226043-f221ae80-9927-11eb-9669-6da157ae5d10.png">

Reply:
I'm fine with the functional change. I feel like 70/30 that you should probably change this to be non-mutating. As you said, the odds are that this is totally fine, but as a pattern that someone else might repeat in the future I think it's best to avoid ... I don't think it reads better enough to justify mutating a prop like this.

Reply:
https://github.com/elastic/kibana/pull/96709/commits/08d279842828c0ddb62bd55aa2daebc02986b834